frontends/uhdm/UhdmAst.cc

```diff
@@ -555,7 +561,14 @@ void UhdmAst::process_module() {
 							  obj_h,
 							  [&](AST::AstNode* node) {
 								  if (node) {
+									if (node->type == AST::AST_INITIAL) {
+										auto insert_it = find_if(current_node->children.begin(), current_node->children.end(),
+												[] (AST::AstNode *node) {return (node->type == AST::AST_ALWAYS);} );
+										// Place AST_INITIAL before AST_ALWAYS if found
+										current_node->children.insert(insert_it, 1, node);
+									} else {
```

Comment:
Correct me, if I'm wrong, but I think this can made design with 2 ``AST_INITIAL`` nodes in the design (and we want only one).

If yes, this part should be moved to ``add_or_replace_child`` and when we are replacing child, we should remove old one and place new one before ``AST_ALWAYS``

Reply:
Yes, You are right. This leads to multiple `AST_INITIAL` nodes in the design. I moved this part to `add_or_replace_child` as a special case for AST_INITIALs. In fact, this piece of code had to be rewritten because we are operating on already existing nodes and we have to move the contents of one node to another ensuring that we have only one AST_INITIAL in the design and  one AST_BLOCK per Initial node. In order to do that a number of specific cases must be handled.